autospec/pkg_integrity.py

```diff
@@ -223,19 +223,21 @@ def head_request(url):
 
 
 def get_signature_url(package_url):
-    if '://pypi.' in package_url[:13]:
+    if 'samba.org' in package_url:
+        return package_url + '.asc'
+    elif '://pypi.' in package_url[:13]:
         return package_url + '.asc'
     elif 'mirrors.kernel.org' in package_url:
         return package_url + '.sig'
     else:
         try:
```

Comment:
This bothers me so much, maybe it's the appropriate moment to re-write the whole `try:` body as:
```python
try:
    iter = (package_url + "." + ext for ext in ("sig", "asc", "sign"))
        for sign_url in iter:
            if head_request(sign_url) in (200, 302):
                return sign_url
except:
    pass
```

Reply:
Agree that is much better, but maybe not as part of this patchset. Feel free to open a PR for that change though.

Reply:
Opened #128. It can wait till this merges so @bryteise wouldn't have to resolve.